packages/astro/src/vite-plugin-build-load-fallback/index.ts

```diff
@@ -0,0 +1,59 @@
+import type { Plugin as VitePlugin, ViteDevServer } from 'vite';
+import type { AstroConfig } from '../@types/astro';
+
+export interface PluginOptions {
+  astroConfig: AstroConfig;
+  viteServer: ViteDevServer;
+}
+
+export function rollupPluginAstroBuildLoadFallback({ astroConfig, viteServer }: PluginOptions): VitePlugin {
```

Comment:
would love some more docs here for future readers: why does this file exist, what does it do, etc.